main.go

```diff
@@ -51,6 +55,34 @@ func connectDb() *sql.DB {
 	return db
 }
 
+func migration(upDown string) {
```

Comment:
This is bad design, do not use a string to accumulate functionality in a single function.
Its better to have two functions `migrateDown` and `migrateUp` for example and extract into another function the shared code (setting up the migration connection for example).
They will be simpler, easier to navigate and less complex (with less `if` statements).

Reply:
If you really need to defer a decision to inside the function execution pass a `boolean` or a `enum` that you could do a switch case or a simple `if thisShouldOccur {` check. Using free formed strings is not a good decision because the function caller will have to inspect the function to know what he could pass as a value to it, also since there is not guard rails in the string any typo could cause the function to have a completely unexpected behavior, lets say the user passes `"ups"` what would the migration function do?